test/Removal.test.ts

```diff
@@ -0,0 +1,162 @@
+    it('should mint a batch of removals without listing any', async () => {
+      const { fifoMarket, removal } = await setupTest();
+      const { supplier } = await hardhat.getNamedAccounts();
+
+      const removalBalances = ['100', '200', '300', '400'];
+      const expectedMarketSupply = '0';
+      const removalVintages = [2018, 2019, 2020, 2021];
+      const listNow = false;
+      const packedData = hardhat.ethers.utils.defaultAbiCoder.encode(
+        ['address', 'bool'],
+        [fifoMarket.address, listNow]
+      );
+
+      const transactionResponse = await removal.mintBatch(
+        supplier,
+        removalBalances.map((balance) =>
+          hardhat.ethers.utils.parseUnits(balance)
+        ),
+        removalVintages,
+        packedData
+      );
+
+      const transactionReceipt = await transactionResponse.wait();
+      const batchMintedEvent = transactionReceipt?.events?.filter((event) => {
+        return event.event === 'BatchMinted';
+      })[0];
+      const tokenIds = (batchMintedEvent?.args?.tokenIds as BigNumber[]).map(
+        (tokenId) => tokenId.toNumber()
+      );
+
+      const balances = await Promise.all(
+        tokenIds.map((tokenId) => {
+          return removal.totalSupply(tokenId);
+        })
+      );
+      balances.forEach((balance, tokenId) => {
+        expect(balance).to.equal(
+          hardhat.ethers.utils.parseUnits(removalBalances[tokenId]).toString()
+        );
+      });
+      // not listed to the fifoMarket
+      const marketTotalSupply = await fifoMarket.numberOfNrtsInQueue();
+      expect(marketTotalSupply).to.equal(
+        hardhat.ethers.utils.parseUnits(expectedMarketSupply).toString()
+      );
+    });
+    it('should mint and list a batch of removals in the same transaction', async () => {
+      const { fifoMarket, removal } = await setupTest();
+
+      const { supplier } = await hardhat.getNamedAccounts();
+
+      const removalBalances = ['100', '200', '300', '400'];
+      const expectedMarketSupply = '1000';
+      const removalVintages = [2018, 2019, 2020, 2021];
+      const listNow = true;
+      const packedData = hardhat.ethers.utils.defaultAbiCoder.encode(
+        ['address', 'bool'],
+        [fifoMarket.address, listNow]
+      );
+      const transactionResponse = await removal.mintBatch(
+        supplier,
+        removalBalances.map((balance) =>
+          hardhat.ethers.utils.parseUnits(balance)
+        ),
+        removalVintages,
+        packedData
+      );
+
+      const transactionReceipt = await transactionResponse.wait();
+      const batchMintedEvent = transactionReceipt?.events?.filter((event) => {
+        return event.event === 'BatchMinted';
+      })[0];
+      const tokenIds = (batchMintedEvent?.args?.tokenIds as BigNumber[]).map(
+        (tokenId) => tokenId.toNumber()
+      );
+
+      const balances = await Promise.all(
+        tokenIds.map((tokenId) => {
+          return removal.totalSupply(tokenId);
+        })
```

Comment:
If the above tokenIds are an empty array this test seems like it could cause a false positive pass